tagging.py

```diff
@@ -0,0 +1,185 @@
+from mutagen import File
+from mutagen.easymp4 import EasyMP4 # used for .m4a and .mp4 files
+from mutagen.flac import FLAC
+from mutagen.easyid3 import EasyID3 # used for .mp3 files
+
+# the AUDIO_TYPES constant dict is used to refer to the mutagen subclasses according to the file format
+#
+# the TAGS_MAPS constant dict is used to refer to the correct tag names for each file format.
+# currently all tags appear the same. that is because the use of EasyID3 and EasyMP4, which mimic the APEv2 format. 
+# regardless, this structure is kept for probable future use.
+AUDIO_TYPES = {"mp3":EasyID3, "mp4":EasyMP4, "flac":FLAC}
+TAGS_MAPS = {'mp3':{'artist':'artist','album':'album','title':'title','track':'tracknumber','year':'date','genre':'genre','albumartist':'albumartist'}, # mp3 tags are EasyID3
+             'mp4':{'artist':'artist','album':'album','title':'title','track':'tracknumber','year':'date','genre':'genre','albumartist':'albumartist'}, # mp4 tags are EasyMP4
+             'flac':{'artist':'artist','album':'album','title':'title','track':'tracknumber','year':'date','genre':'genre','albumartist':'albumartist'}}
+
+
+class AudioFile:
+    def __init__(self,filename):
+        audiotype = File(filename).mime[0].split("/")[1]    # determine the file format
+        self.meta = AUDIO_TYPES[audiotype](filename)        # create the tags instance of the appropriate mutagen subclass
+        self.tag_map = TAGS_MAPS[audiotype]                 # used to call the tags of any format by the same names
+        
+    def tag_entry(self,tags_dict):
+        '''
+            takes a dict of tags. notice the tags_dict keys should match tags2func keys.
+            deletes any old tags on the file and sets the new tags.
+        '''
+        tag2func = {'artist':self.artist,'album':self.album,'title':self.title,'track':self.track,'year':self.year,'genre':self.genre,'albumartist':self.albumartist}
+        self.delete_tags()
+        for tagname in tags_dict:
+            tag2func[tagname](tags_dict[tagname])
+        self.meta.save()
+    
+    def delete_tags(self):
+        '''
+            remove all the tags from the file.
+            this leaves the tags as an empty dict. 
+        '''
+        self.meta.clear()
+        self.meta.save()
+    
+    def artist(self,entry=None):
```

Comment:
bad naming in all of setting methods
"set_artist/name/album_tag" would be a lot better